examples/async_echo.py

```diff
@@ -1,23 +1,28 @@
 import asyncio
-import os
 import pathlib
 
 import pycape
-from pycape.experimental import cli
 
+parent_dir = pathlib.Path(__file__).parent.absolute()
+echo_func_file = parent_dir / "echo.json"
+token_file = parent_dir / "capedocs.token"
+function_owner = "capedocs"
 
-async def main(cape: pycape.Cape, deploy_path: os.PathLike, echo: str) -> str:
-    function_ref = await cli.deploy(deploy_path)
+
+async def main(
+    cape: pycape.Cape, function: pycape.FunctionRef, token: pycape.Token, echo: str
+) -> str:
     echo_arg = echo.encode()
-    echo_enc = await cape.encrypt(echo_arg, token=function_ref.token)
-    async with cape.function_context(function_ref):
+    echo_enc = await cape.encrypt(echo_arg, username=function.user)
+    async with cape.function_context(function, token):
         result = await cape.invoke(echo_enc)
     return result.decode()
 
 
 if __name__ == "__main__":
-    echo_path = pathlib.Path(__file__).parent.absolute() / "echo"
 
     cape = pycape.Cape()
-    echo = asyncio.run(main(cape, echo_path, "Welcome to Cape."))
+    fref = cape.function(echo_func_file)
+    token = cape.token(token_file)
```

Comment:
Not following why you are passing a file name to this method.

Reply:
the argument is duck-typed, the method's [type signature](https://github.com/capeprivacy/pycape/pull/113/files#diff-036ba23fbfebbf74e74e3d14c94feb8e5804dd76de635597f24d0a9a4dcfa1e5R447) enumerates the possible types